Commit fabee26a41fe "[pci] Add helpers to request/release memory and I/O
regions" (upstream be9d2e8927ce) introduced pci_{request,release}_mem_regions()
function helpers, which then utilised in 9ef8b4631270e "[netdrv] ethernet/intel:
Use pci_(request|release)_mem_regions" (upstream 56d766d64ca5c).

Backport change: add implementation of pci_{request,release}_mem_regions() from
"[pci] Add helpers to request/release memory and I/O regions" to
i40e_backport_compat.h.

Index: src/drivers/net/ethernet/intel/i40e/i40e_backport_compat.h
===================================================================
--- src.orig/drivers/net/ethernet/intel/i40e/i40e_backport_compat.h	2017-04-27 19:43:48.091036921 +0200
+++ src/drivers/net/ethernet/intel/i40e/i40e_backport_compat.h	2017-04-27 19:45:56.784506772 +0200
@@ -2,6 +2,7 @@
 #define _RH_I40E_BACKPORT_COMPAT_H_
 
 #include <net/udp_tunnel.h>
+#include <linux/pci.h>
 
 
 #define NETIF_F_TSO_MANGLEID 0
@@ -45,4 +46,18 @@
 }
 
 
+static inline int
+pci_request_mem_regions(struct pci_dev *pdev, const char *name)
+{
+	return pci_request_selected_regions(pdev,
+			pci_select_bars(pdev, IORESOURCE_MEM), name);
+}
+
+static inline void
+pci_release_mem_regions(struct pci_dev *pdev)
+{
+	return pci_release_selected_regions(pdev,
+			pci_select_bars(pdev, IORESOURCE_MEM));
+}
+
 #endif /* #ifndef _RH_I40E_BACKPORT_COMPAT_H_ */
